app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find_by(id: params[:user_id])
-    @posts = Post.where(author_id: params[:user_id])
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments, :likes)
   end
```

Comment:
- [ ] Good job  trying to solve the n+1 problem. However, as required for the project, kindly ensure that you add screenshots of the logs before and after the fix to the PR description as right now they are missing there.